src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -420,6 +420,37 @@ static void create_stub_global_symbol(
     !add_failed, "caller should have checked symbol not already in table");
 }
 
+/// Find any incomplete ancestor of a given class
+/// \param start_class_id: class to start searching from
+/// \param symbol_table: global symbol table
+/// \param class_hierarchy: global class hierarchy
+/// \return first incomplete ancestor encountered,
+///   including start_class_id itself.
+static irep_idt get_any_incomplete_ancestor(
+  const irep_idt &start_class_id,
+  const symbol_tablet &symbol_table,
+  const class_hierarchyt &class_hierarchy)
+{
+  std::vector<irep_idt> classes_to_check;
```

Comment:
Might be simpler to read what's going on with
```C++
idst parents = class_heirachy.get_parents_trans(start_class_id)
std::find(
  parents.begin(), 
  parents.end(), 
  [](const irep_idt &to_check) { 
    return symbol_table.lookup_ref(to_check).type.get_bool(ID_incomplete_class);
});
```

Reply:
I think that would be a bit shaky, as if we ever created a hierarchy where incomplete classes inherited from one another (not currently possible but conceivable) then this might return an incomplete class with incomplete children that are closer to the one we're asking about. That wouldn't be wrong, but it would be surprising.

I've added an explanatory comment instead.